File: app/src/main/java/br/com/accenture/santander/wallacebaldenebre/ui/login/LoginContract.java
package br.com.accenture.santander.wallacebaldenebre.ui.login;

import br.com.accenture.santander.wallacebaldenebre.ui.base.MvpPresenter;
import br.com.accenture.santander.wallacebaldenebre.ui.base.MvpView;

public interface LoginContract {
    interface View extends MvpView {
        void makeLogin();

        void setUp();
    }

    interface Presenter<V extends MvpView> extends MvpPresenter<V> {
        void makeLogin(LoginActivity login, String user, String password);
    }
}

File: BankApp/app/src/main/java/br/com/rms/bankapp/ui/login/LoginContract.kt
package br.com.rms.bankapp.ui.login

import androidx.lifecycle.Lifecycle
import androidx.lifecycle.LifecycleOwner
import br.com.rms.bankapp.base.mvp.BaseContract
import br.com.rms.bankapp.utils.validations.ValidationException

interface LoginContract : BaseContract.View{

    interface View : BaseContract.View, LifecycleOwner {
        override fun getLifecycle(): Lifecycle
        fun validateError(e: ValidationException)
        fun showLoader()
        fun hideLoader()
        fun showErrorMessage(error_message_request_login: Int)
        fun loginSuccess()
        fun setUser(user: String?)

    }

    interface Presenter : BaseContract.Presenter{
        fun login(userLogin: String, userPassword: String)
    }
}